docs/API.md

```diff
@@ -824,6 +834,115 @@ __Example__
+|        | ``InsufficientDataException`` : Thrown to indicate that reading given InputStream gets EOFException before reading given length. |
+|        | ``IOException`` : upon connection error.            |
+|        | ``InvalidKeyException`` : upon an invalid access key or secret key.           |
+|        | ``NoResponseException`` : upon no response from server.            |
+|        | ``org.xmlpull.v1.XmlPullParserException`` : upon parsing response XML.            |
+|        | ``ErrorResponseException`` : upon unsuccessful execution.            |
+|        | ``InternalException`` : upon internal library error.        |
+
+
+__Example__
+
+
+```java
+ try {
+      NotificationConfiguration notificationConfiguration = minioClient.getBucketNotification("my-bucketname");
+
+      // Add a new SQS configuration.
+      List<QueueConfiguration> queueConfigurationList = notificationConfiguration.queueConfigurationList();
+      QueueConfiguration queueConfiguration = new QueueConfiguration();
+      queueConfiguration.setQueue("arn:minio:sqs::1:webhook");
+
+      List<EventType> eventList = new LinkedList<>();
+      eventList.add(EventType.OBJECT_CREATED_PUT);
+      eventList.add(EventType.OBJECT_CREATED_COPY);
+      queueConfiguration.setEvents(eventList);
+
+      Filter filter = new Filter();
+      filter.setPrefixRule("images");
+      filter.setSuffixRule("pg");
+      queueConfiguration.setFilter(filter);
+
+      queueConfigurationList.add(queueConfiguration);
+      notificationConfiguration.setQueueConfigurationList(queueConfigurationList);
+
+      // Set updated notification configuration.
+      minioClient.setBucketNotification("my-bucketname", notificationConfiguration);
+      System.out.println("Bucket notification is set successfully");
+    } catch (MinioException e) {
+      System.out.println("Error occurred: " + e);
+    }
+```
+
+  <a name="getBucketNotification"></a>
+### getBucketNotification(String bucketName)
+`public NotificationConfiguration getBucketNotification(String bucketName)`
+
+Get bucket notification configuration.
+
+[View Javadoc](http://minio.github.io/minio-java/io/minio/MinioClient.html#getBucketNotification-java.lang.String-)
+
+__Parameters__
+
+|Param   | Type   | Description  |
+|:--- |:--- |:--- |
+| ``bucketName``  | _String_  | Name of the bucket.  |
+
+| Return Type	  | Exceptions	  |
+|:--- |:--- |
+|  ``NotificationConfiguration``:  NotificationConfiguration    | Listed Exceptions: |
+|        | ``InvalidBucketNameException`` : upon invalid bucket name. |
+|        | ``InvalidObjectPrefixException`` : upon invalid object prefix. |
+|        | ``NoSuchAlgorithmException`` : upon requested algorithm was not found during signature calculation.  |
+|        | ``InsufficientDataException`` : Thrown to indicate that reading given InputStream gets EOFException before reading given length. |
```

Comment:
The same comment here.

This is documentation, it needs to be meaningful text, imho.
So, what is `given InputStream` from the perspective of the end-user? 
Why do we say `given`? This is not an input parameter end-user provides.
We should not reveal any internal parameters when we explain the meaning of return values.

You also need a space in between `EOF` and `Exception`. Even it would be better to call it something like `"End of file exception"`.

Reply:
modified